packages/my-near-wallet/src/lib/my-near-wallet.ts

```diff
@@ -160,6 +160,32 @@ const MyNearWallet: WalletBehaviourFactory<
       throw new Error(`Method not supported by ${metadata.name}`);
     },
 
+    async signMessage({ message, nonce, recipient, callbackUrl, state }) {
+      logger.log("sign message", { message });
+
+      const account = _state.wallet.account();
+
+      if (!account) {
+        throw new Error("Wallet not signed in");
+      }
+      const locationUrl =
+        typeof window !== "undefined" ? window.location.href : "";
+
+      const url = callbackUrl || locationUrl;
+
+      if (!url) {
+        throw new Error(`The callbackUrl is missing for ${metadata.name}`);
+      }
+
+      const encodedUrl = encodeURIComponent(url);
+
+      window.location.replace(
+        `${params.walletUrl}/sign-message?message=${message}&nonce=${nonce}&state=${state}&recipient=${recipient}&callbackUrl=${encodedUrl}`
```

Comment:
After some testing, I found out that depending on what value for `nonce` we pass to the URL it might break the URL structure.

For example trying to sign a message with a nonce like this:

```ts
const nonce = Buffer.from(Array.from(Array(32).keys()));
```

We see this error on the wallet:
![image](https://github.com/near/wallet-selector/assets/95851345/0587f136-2993-4441-acb9-41d7809fc5af)

But if the `nonce` is created from a 32-character long string then the wallet shows the correct UI:
```ts
const nonce = Buffer.from("abcdefghijklmnopqrstuvwxyz123456");
```
![image](https://github.com/near/wallet-selector/assets/95851345/60cf6f47-943a-4bbf-825c-2c7f68afadf5)